service-app/internal/httpclient/request.go

```diff
@@ -0,0 +1,66 @@
+package httpclient
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/ministryofjustice/opg-scanning/config"
+	"github.com/ministryofjustice/opg-scanning/internal/logger"
+)
+
+type HttpClient struct {
+	HttpClient *http.Client
+	Config     config.Config
+	Logger     logger.Logger
+}
+
+func NewHttpClient(config config.Config, logger logger.Logger) *HttpClient {
+	return &HttpClient{
+		HttpClient: &http.Client{
+			Timeout: time.Duration(config.HTTP.Timeout) * time.Second,
+		},
+		Config: config,
+		Logger: logger,
+	}
+}
+
+func (r *HttpClient) HTTPRequest(url, method string, payload []byte, headers map[string]string) ([]byte, error) {
+	// Log the request details before sending it
+	r.Logger.Info(fmt.Sprintf("Sending request to URL: %s - Payload:\n%s", url, payload))
```

Comment:
Shouldn't log the payload here: for document requests it's going to contain the base64 of an entire PDF, and we prefer not to log personally-identifiable information (which the PDF will be full of).

Reply:
Agreed, will remove the payload part.

Reply:
Resolved in SSM-23